client/me/account/main.jsx

```diff
@@ -138,15 +145,29 @@ const Account = createReactClass( {
 	},
 
 	getUserSetting( settingName ) {
-		return this.props.userSettings.getSetting( settingName );
+		return (
+			get( this.props.unsavedUserSettings, settingName ) ??
+			this.getUserOriginalSetting( settingName )
+		);
 	},
 
 	getUserOriginalSetting( settingName ) {
-		return this.props.userSettings.getOriginalSetting( settingName );
+		return get( this.props.userSettings, settingName );
+	},
+
+	hasUnsavedUserSetting( settingName ) {
+		return has( this.props.unsavedUserSettings, settingName );
```

Comment:
Is `has()` preferred than `this.props.unsavedUserSettings?.hasOwnProperty( settingName )` for a reason? 

FWIW we also have other instances of `has()` here, consider this question applicable to them, too.

Reply:
See my comment here: https://github.com/Automattic/wp-calypso/pull/49760#discussion_r571189849 – also applies to `has`.